### PR TITLE
Migrated scheduler files `comparer.go`, `dumper.go`, `node_tree.go` to structured logging

### PR DESCRIPTION
**Pull request by shivanshuraj1333**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Migrated `pkg/scheduler/internal/cache/debugger/comparer.go`, `pkg/scheduler/internal/cache/debugger/dumper.go`, and `pkg/scheduler/internal/cache/node_tree.go` to structured logging
#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/issues/105841

#### Special notes for your reviewer:
Added missing migration points, fixed previously migrated logs
#### Does this PR introduce a user-facing change?

```release-note
Migrated scheduler files `comparer.go`, `dumper.go`, `node_tree.go` to structured logging
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/internal/cache/node_tree.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (nt *nodeTree) addNode(n *v1.Node) {
 	if na, ok := nt.tree[zone]; ok {
 		for _, nodeName := range na {
 			if nodeName == n.Name {
-				klog.Warningf("node %q already exist in the NodeTree", n.Name)
+				klog.InfoS("Node already exist in the NodeTree", "node", klog.KObj(n))
```
[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/node_tree.go:56]: done thanks!

File: pkg/scheduler/internal/cache/debugger/dumper.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func (d *CacheDumper) dumpSchedulingQueue() {
 	for _, p := range pendingPods {
 		podData.WriteString(printPod(p))
```
[the remark this answers is not in the document]
Comment (pohly) [reply pkg/scheduler/internal/cache/debugger/dumper.go:58]: @shivanshu1333: do you intend to implement such a formatter as part of this PR?

[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/debugger/dumper.go:58]: Yes, actually I've some doubts I need your help to figure out the right way to do this, I had posted a query in slack for the same https://kubernetes.slack.com/archives/C020CCMUEAX/p1636304943054700

**Comment by shivanshu1333** [comment]
The unit test to check the behaviour for `formatter.go` is https://gist.github.com/shivanshu1333/3e7b7cda9fad55c2065900cd30f75781
If needed, I can add the unit test in the project itself

**Comment by shivanshu1333** [comment]
cc @pohly @serathius @damemi

**Comment by shivanshu1333** [comment]
Please let me know if any change is required, and also let’s think about adding the above unit test as well, please guide me about possibilities of introducing a unit test for `formatter.go`

**Review by pohly — CHANGES_REQUESTED** [review]

**Comment by shivanshu1333** [comment]
> Your formatter looks (mostly) good to me and we should come back to it once we know how to emit multi-line strings in text format, but for now we should keep the original klog.Info(d.printNodeInfo(name, nodeInfo)) and
> klog.Infof("Dump of scheduling queue:\n%s", podData.String())

@pohly  Does this mean that for the scope of this migration PR, let's not worry about the formatter in this PR and keep the original things as it is, and start new efforts (PR) to continue formatting work?

**Comment by pohly** [comment]
Yes. It means the migration won't be complete, but that's better than migrating and producing output that is worse than what it is without the migration.

**Comment by shivanshu1333** [comment]
Created https://github.com/kubernetes/kubernetes/issues/106276 to take care of custom formatter, I think now this PR is unblocked and we can work on the mentioned issue separately.
Looking for LGTM and APPROVED label on this

File: pkg/scheduler/internal/cache/debugger/dumper.go [hunk 3]
```diff
@@ -44,9 +44,9 @@ func (d *CacheDumper) DumpAll() {
 // dumpNodes writes NodeInfo to the scheduler logs.
 func (d *CacheDumper) dumpNodes() {
 	dump := d.cache.Dump()
-	klog.Info("Dump of cached NodeInfo")
+	klog.InfoS("Dump of cached NodeInfo")
 	for name, nodeInfo := range dump.Nodes {
-		klog.Info(d.printNodeInfo(name, nodeInfo))
+		klog.InfoS(d.printNodeInfo(name, nodeInfo))
```
Comment (pohly) [inline pkg/scheduler/internal/cache/debugger/dumper.go:49]: Passing a message string with line breaks to `InfoS` has the same problem as passing it a value: it gets printed as quoted string on a single line.

You have to keep using `klog.Info` here.

File: pkg/scheduler/internal/cache/debugger/dumper.go [hunk 4]
```diff
@@ -57,7 +57,7 @@ func (d *CacheDumper) dumpSchedulingQueue() {
 	for _, p := range pendingPods {
 		podData.WriteString(printPod(p))
 	}
-	klog.Infof("Dump of scheduling queue:\n%s", podData.String())
+	klog.InfoS("Dump of scheduling queue", "podData", podData)
```
Comment (pohly) [inline pkg/scheduler/internal/cache/debugger/dumper.go:60]: ... and here.

**Review by pohly — CHANGES_REQUESTED** [review]

File: pkg/scheduler/internal/cache/debugger/dumper.go (hunk 3 above)
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/debugger/dumper.go]: Okay, done!

File: pkg/scheduler/internal/cache/debugger/dumper.go (hunk 4 above)
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/debugger/dumper.go]: done

**Comment by Huang-Wei** [comment]
/approve

Leaving /lgtm to @pohly.

**Review by pohly — APPROVED** [review]
/lgtm

**Pull request merged**